packages/docusaurus-init/templates/bootstrap/package.json

```diff
@@ -16,10 +16,13 @@
   "dependencies": {
     "@docusaurus/core": "2.0.0-alpha.73",
     "@docusaurus/preset-bootstrap": "2.0.0-alpha.73",
-    "@mdx-js/react": "^1.5.8",
-    "classnames": "^2.2.6",
+    "@mdx-js/react": "^1.6.21",
+    "@svgr/webpack": "^5.5.0",
+    "clsx": "^1.1.1",
+    "file-loader": "^6.2.0",
     "react": "^17.0.1",
-    "react-dom": "^17.0.1"
+    "react-dom": "^17.0.1",
+    "url-loader": "^4.1.1"
```

Comment:
You should not use this with webpack v5

Reply:
Ideally asset modules should be used instead, correct?

Reply:
Yes, it has more features :star: and in future even more

Reply:
Thanks, will look at how much work is needed to change that.

Is it a required change? Maybe we should do it in another PR later?

Reply:
@alexander-akait I tried to replace the file/url loaders with asset modules. 

Overall it seems to work great, just had to add a few loaders with resourceQuery ?asset at the very beginning of the rules to emulate the old behavior of inline loaders.

However I have one error left that I'm not able to solve:

> Conflict: Multiple assets emit different content to the same filename assets/ideal-img/datagit.100.png

It is related to our "ideal-image" + "lqip-loader" (still using file-loader internally, used to resize images and produce dataUri previews). 
I'm not sure how to migrate this code to assets or if it's even something I should do.

This plugin is using on both server/client config, but it should only emit files on the client config so I guess there should be no conflict due to using it in 2 configs. I'm not 100% sure my current config is correct nor to understand the `javascript/auto` thing.

Can you help me figure this out, please?
I have a branch here: https://github.com/facebook/docusaurus/commits/slorber/webpack5-asset-modules
A single commit. to check: https://github.com/facebook/docusaurus/commit/2f21d306bdd4d286cc5d25c81adaea2fc77f0474

The relevant files are probably:
- https://github.com/facebook/docusaurus/blob/slorber/webpack5-asset-modules/packages/docusaurus-plugin-ideal-image/src/index.ts
- https://github.com/facebook/docusaurus/blob/slorber/webpack5-asset-modules/packages/lqip-loader/src/index.js


Anyway, I don't think this asset module refactor absolutely needs to be part of this PR and block it, but if I can find a solution soon I'd be happy to include it.

Reply:
So it seems I only have this error in dev (not prod) and it's not related to client/server config both trying to emit the same file.

In dev the filename does not have a hash (not sure why).

```js
 {
              use: [
                require.resolve('@docusaurus/lqip-loader'),
                {
                  loader: require.resolve('@docusaurus/responsive-loader'),
                  options: {
                    emitFile: !isServer, // don't emit for server-side rendering
                    disable: !isProd,
                    adapter: require('@docusaurus/responsive-loader/sharp'),
                    name: isProd
                      ? 'assets/ideal-img/[name].[hash:hex:7].[width].[ext]'
                      : 'assets/ideal-img/[name].[width].[ext]',
                    ...options,
                  },
                },
              ],
            },
```

Adding a hash in dev filename fixes the problem. I don't even know why we need different dev/prod filenames, so I'm going to move on and just add a hash.

However, this problem only surfaced after migrating some existing file/url loaders to asset modules, I have no idea why :o

Reply:
oh, actually this was a totally legit warning... just figured out we had 2 images with the exact same basename so using the hash is mandatory